metrics/prometheus.go

```diff
@@ -1211,7 +1211,27 @@ func (c *PrometheusCollector) collectContainersInfo(ch chan<- prometheus.Metric)
 				)
 			}
 		}
+		if c.includedMetrics.Has(container.AppMetrics) {
+			for metricLabel, v := range stats.CustomMetrics {
+				for _, metric := range v {
+					clabels := make([]string, len(rawLabels), len(rawLabels)+len(metric.Labels))
+					cvalues := make([]string, len(rawLabels), len(rawLabels)+len(metric.Labels))
+					copy(clabels, labels)
+					copy(cvalues, values)
+					for label, value := range metric.Labels {
+						if label == "__name__" {
```

Comment:
why are we special-casing __name__?

Reply:
```
resources are being tracked.
panic: "__name__" is not a valid label name

goroutine 568 [running]:
github.com/google/cadvisor/vendor/github.com/prometheus/client_golang/prometheus.MustNewConstMetric(...)
	/go/src/github.com/google/cadvisor/vendor/github.com/prometheus/client_golang/prometheus/value.go:102
github.com/google/cadvisor/metrics.(*PrometheusCollector).collectContainersInfo(0xc000110140, 0xc0007b6d20)
	/go/src/github.com/google/cadvisor/metrics/prometheus.go:1229 +0x2044
github.com/google/cadvisor/metrics.(*PrometheusCollector).Collect(0xc000110140, 0xc0007b6d20)
	/go/src/github.com/google/cadvisor/metrics/prometheus.go:1078 +0x89
github.com/google/cadvisor/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
	/go/src/github.com/google/cadvisor/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x19d
created by github.com/google/cadvisor/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/go/src/github.com/google/cadvisor/vendor/github.com/prometheus/client_golang/prometheus/registry.go:526 +0xe12
```

Reply:
maybe it's better to do it here?
https://github.com/google/cadvisor/pull/2262/files/70e91edc25b101bec16138f4a2290b1427aa6a05#diff-4d5a6aee7c1d41b24004ab60e8074b32R187

Reply:
yeah, I would prefer doing any filtering at collection time.